app/index.js

```diff
@@ -19,15 +19,16 @@ module.exports = yeoman.generators.Base.extend({
 			type: 'input',
 			name: 'componentName',
 			message: 'How do you want to name your class?',
-			default: 'Modal',
+			default: 'Select Input',
 			validate: function(input) {
 				if (!input) {
-					return 'You must provide a class name.';
+					return 'You must provide a class name. Names should be '
+					'capitalized and separated by spaces.';
 				}
-				if (!/^[^_\-\s\d][^_\-\s]*$/.test(input)) {
-					return 'Invalid class name. Class names can\'t contain whitespace or ' +
-					'any of the following characters: "-_". Also, class names can\'t ' +
-					'start with digits.';
+				if (!/^[^_\-\s\d](?:[^_\-\s]| )*$/.test(input)) {
+					return 'Invalid component name. Component names can\'t ' +
```

Comment:
It'd be best to keep saying `class name` instead of `component name`, since the generated class doesn't need to be extending from `Component`. In fact we ask afterwards which super class you'd like to use (which could even be none btw), so I think it could be confusing to say `component` here.

You probably did this so people could use spaces when typing the name, but if we keep calling it `class name` we can make it explicit that it should use CamelCase, and split that into spaces instead of doing the opposite.